### PR TITLE
Use Date type in TPC-H connector

### PR DESCRIPTION
**Pull request by majetideepak**
Change the type of date columns from VARCHAR to DATE.

Resolves https://github.com/facebookincubator/velox/issues/2618

### CODE REVIEW

File: velox/connectors/tpch/tests/TpchConnectorTest.cpp [hunk 1]
```diff
@@ -251,6 +251,18 @@ TEST_F(TpchConnectorTest, join) {
...
+TEST_F(TpchConnectorTest, orderDateCount) {
+  auto plan = PlanBuilder()
+                  .tableScan(Table::TBL_ORDERS, {"o_orderdate"}, 1)
+                  .filter("o_orderdate = '1992-01-01'::DATE")
+                  .singleAggregation({}, {"count(1)"})
+                  .planNode();
+
+  auto output = getResults(plan, {makeTpchSplit()});
+  // Match with Java output.
+  EXPECT_EQ(621, output->childAt(0)->asFlatVector<int64_t>()->valueAt(0));
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/connectors/tpch/tests/TpchConnectorTest.cpp:263]: Why would it be worth checking the actual date value? The equality filter checks for the value anyway right?
But the count will help ensure all the expected dates are parsed correctly.

[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/connectors/tpch/tests/TpchConnectorTest.cpp]: With the support for decimal scaleFactor, I was able to add a simpler test.
I test both the actual date value and the count now.
There are other tests as well for Date value check.

**Comment by majetideepak** [comment]
@pedroerp can you please review this PR?
I removed some redundant scaleFactor non-negative checks inside `genTpchXxx` methods since they all call `getRowCount` at the very beginning which checks for non-negative values also.
I added a check to `DBGenIterator` for API correctness sake.

**Review by pedroerp — APPROVED** [review]
Nice and clean! Thanks for looking into this :)

**Pull request closed without merging**